library/account/src/main/kotlin/org/cru/godtools/account/GodToolsAccountManager.kt

```diff
@@ -26,20 +28,28 @@ import org.ccci.gto.android.common.base.Ordered
 import org.cru.godtools.account.provider.AccountProvider
 import org.cru.godtools.account.provider.AuthenticationException
 import org.cru.godtools.api.UserApi
+import org.cru.godtools.base.BaseModule.IS_CONNECTED_STATE_FLOW
 
 @Singleton
 @OptIn(ExperimentalCoroutinesApi::class)
 class GodToolsAccountManager @VisibleForTesting internal constructor(
     @get:VisibleForTesting
     internal val providers: List<AccountProvider>,
     private val userApi: Lazy<UserApi>,
+    @Named(IS_CONNECTED_STATE_FLOW)
+    private val isConnected: StateFlow<Boolean>,
     coroutineScope: CoroutineScope = CoroutineScope(SupervisorJob()),
 ) {
     @Inject
     internal constructor(
         providers: Set<@JvmSuppressWildcards AccountProvider>,
         userApi: Lazy<UserApi>,
-    ) : this(providers.sortedWith(Ordered.COMPARATOR), userApi)
+        @Named(IS_CONNECTED_STATE_FLOW)
+        isConnected: StateFlow<Boolean>,
+        coroutineScope: CoroutineScope = CoroutineScope(SupervisorJob()),
+    ) : this(providers.sortedWith(Ordered.COMPARATOR), userApi, isConnected)
+
+    private val isConnectedFlow = isConnected.stateIn(coroutineScope, SharingStarted.Eagerly, false)
```

Comment:
is there a reason you are recreating the `StateFlow` you are already wiring into the `GodToolsAccountManager`?

Reply:
I'll restate my previous review comment:

> I'm actually not a fan of using a StateFlow like this, by collecting eagerly when the AccountManager is created you will always be monitoring the online status even if the user never touches the login screen.
>
> There is no need to continuously monitor the online state, it should just be a single check to see if the user is online when an error is encountered

Reply:
Sorry pulled in too many directions and overlooked that part.  My thought on doing this was to place the flow within the same coroutine scope.  Please correct me if I have the wrong understanding of this.

Reply:
@gyasistory let's talk through this at some point today, there's a few things going on here that would be easier to talk through vs type out